### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by iros**
Test infrastructure + a few test files, one for unit tests (using formatters) and one testing the actual react component (taco table).
Uses Karma + Mocha + Chai, and the Chrome karma tester.

### CODE REVIEW

File: tests/Formatters.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  describe('safeFormat', function () {
+    it('should return null when passed a null', function () {
+      assert.equal(Formatters.safeFormat(null, identity), null);
+    });
+
+    it('should return undefined when passed a undefined', function () {
+      assert.equal(Formatters.safeFormat(undefined, identity), undefined);
+    });
+
+    it('should return value when passed a value that is safe', function () {
```
Comment (pbeshai) [inline tests/Formatters.js]: should return at **formatted** value

File: tests/TacoTable.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    const tableNode = (
+      <TacoTable {...props} />
+    );
+    const table = renderIntoDocument(
+      tableNode, root
+    );
+    return findRenderedComponentWithType(table, TacoTable);
+  }
+
+  const columnDefinition = [
```
Comment (pbeshai) [inline tests/TacoTable.js]: columnDefinition**s** or simply **columns**

File: tests/TacoTable.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      foundTable = defineTable(props);
+    });
+    it('should have a table rendered', function () {
+      expect(foundTable).to.be.ok;
+    });
+    it('should have two columns', function () {
+      assert.equal(foundTable.props.columns.length, 2);
+    });
+    it('should have 5 rows', function () {
+      assert.equal(foundTable.props.data.length, 5);
```
Comment (pbeshai) [inline tests/TacoTable.js]: These tests seem to only check that the props was set, but really we'd like to see if there are 5 TRs in the <tbody> with 2 <TD>s in each

File: tests/TacoTable.js (hunk 2 above)
Comment (iros) [inline tests/TacoTable.js]: +1 columns.

File: tests/Formatters.js (hunk 1 above)
Comment (iros) [inline tests/Formatters.js]: Updating all test names to reflect where a formatted value should be returned.

File: tests/TacoTable.js (hunk 3 above)
Comment (iros) [inline tests/TacoTable.js]: Adding actual dom tests.

**Comment by iros** [comment]
Updated @pbeshai. ping!

**Comment by pbeshai** [comment]
+1 Thanks @iros!

**Pull request merged**